api/src/main/java/com/velocitypowered/api/util/GameProfile.java

```diff
@@ -8,24 +8,36 @@
 /**
  * Represents a Mojang game profile. This class is immutable.
  */
-public final class GameProfile {
+public final class GameProfile implements Identifiable {
 
-  private final String id;
-  private final String name;
+  private final UUID id;
+  private final String undashedId, name;
   private final List<Property> properties;
 
-  public GameProfile(String id, String name, List<Property> properties) {
-    this.id = Preconditions.checkNotNull(id, "id");
-    this.name = Preconditions.checkNotNull(name, "name");
-    this.properties = ImmutableList.copyOf(properties);
+  public GameProfile(UUID id, String name, List<Property> properties) {
+    this(Preconditions.checkNotNull(id, "id"), UuidUtils.toUndashed(id),
+        Preconditions.checkNotNull(name, "name"), ImmutableList.copyOf(properties));
   }
 
-  public String getId() {
-    return id;
+  public GameProfile(String undashedId, String name, List<Property> properties) {
+    this(UuidUtils.fromUndashed(Preconditions.checkNotNull(undashedId, "undashedId")), undashedId,
+        Preconditions.checkNotNull(name, "name"), ImmutableList.copyOf(properties));
   }
 
-  public UUID idAsUuid() {
```

Comment:
Breaking API change. Please revert.

Reply:
`idAsUuid` sounded a bit unclear to me and I didn't know if you wanted to keep the API unchanged or not but I will revert it.

Reply:
Should I deprecate this method in favor of `getUniqueId` or remove `getUniqueId` and thus probably `Identifiable` by the way, or just keep both methods without deprecating ?

Reply:
Deprecation would be acceptable if we keep `getUniqueId()`, but I'm still not sold on the need for the `Identifiable` interface.